src/main/webapp/app/exercises/programming/assess/code-editor-tutor-assessment-container.component.ts

```diff
@@ -371,7 +371,7 @@ export class CodeEditorTutorAssessmentContainerComponent implements OnInit, OnDe
 
     private handleFeedback(): void {
         // Setup automatic feedback
-        this.automaticFeedback = this.automaticResult?.feedbacks!;
+        this.automaticFeedback = this.automaticResult?.feedbacks! ?? [];
```

Comment:
does this really work? I guess you would need to leave out the `!` operator after `feedbacks`

Reply:
```suggestion
        this.automaticFeedback = this.automaticResult?.feedbacks ?? [];
```

Reply:
you are right, this is not needed and can be removed. Good catch! 👍

Reply:
I think it still works but removing the ! is cleaner.